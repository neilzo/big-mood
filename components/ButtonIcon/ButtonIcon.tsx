import React from 'react';
import { View, TouchableWithoutFeedback, StyleSheet } from 'react-native';

import Icon from '../Icon/Icon';

interface Props {
  onPress: () => void;
  icon: string;
  size?: number;
}
const ButtonIconComponent = ({ icon, onPress, size = 25 }: Props) => (
  <TouchableWithoutFeedback onPress={onPress}>
    <View style={styles.container}>
      <Icon name={icon} size={size} />
    </View>
  </TouchableWithoutFeedback>
);

const styles = StyleSheet.create({
  container: {}
});

export default ButtonIconComponent;
