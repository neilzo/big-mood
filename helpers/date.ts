import moment from 'moment';

const getPrettyDate = (date: Date) => moment(date).format('MMM Do YYYY');

const getPrettyTime = (date: Date) => moment(date).format('h:mma');

export default { getPrettyDate, getPrettyTime };
