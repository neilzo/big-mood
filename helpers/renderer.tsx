import 'react-native';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

export const createComponent = (component: any) => {
  return renderer.create(component);
};

// export const createComponentWithRedux = (children: any, props = {}) => {
//   return renderer.create(<Provider {...props}>{children}</Provider>);
// };
