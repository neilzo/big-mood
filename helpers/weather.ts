interface WeatherResults {
  weather: Array<{ main: string; icon: string }>;
  main: {
    temp: number;
  };
}

const kelvinToFarenheit = (kelvin: number) => {
  return Math.round(kelvin * (9 / 5) - 459.76);
};

const computeOverallWeather = (weatherResults: WeatherResults) => ({
  icon: weatherResults.weather[0].icon,
  description: weatherResults.weather[0].main,
  temperature: kelvinToFarenheit(weatherResults.main.temp),
});

export { kelvinToFarenheit, computeOverallWeather };
