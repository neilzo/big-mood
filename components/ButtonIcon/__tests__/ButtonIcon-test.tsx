import React from 'react';
import ButtonIcon from '../ButtonIcon';
import { createComponent } from '../../../helpers/renderer'; // todo resolve imports to avid that ugly thang

jest.mock('../../Icon/Icon', () => 'Icon');

const mockProps = {
  onPress: () => {},
  icon: 'rocket'
};

describe('ButtonIcon', () => {
  it('renders correctly', () => {
    expect(
      createComponent(<ButtonIcon {...mockProps} />).toJSON()
    ).toMatchSnapshot();
  });

  it('renders correctly with size', () => {
    expect(
      createComponent(<ButtonIcon {...mockProps} size={50} />).toJSON()
    ).toMatchSnapshot();
  });
});
