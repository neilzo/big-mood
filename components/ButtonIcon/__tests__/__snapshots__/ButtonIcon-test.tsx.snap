// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon renders correctly 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
>
  <Icon
    name="rocket"
    size={25}
  />
</View>
`;

exports[`ButtonIcon renders correctly with size 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
>
  <Icon
    name="rocket"
    size={50}
  />
</View>
`;
